pkg/apis/cosigned/v1alpha1/clusterimagepolicy_validation_test.go

```diff
@@ -397,6 +397,61 @@ func TestAuthoritiesValidation(t *testing.T) {
 				},
 			},
 		},
+		{
+			name:      "Should pass when source oci is present",
+			expectErr: false,
+			policy: ClusterImagePolicy{
+				Spec: ClusterImagePolicySpec{
+					Images: []ImagePattern{{Regex: ".*"}},
+					Authorities: []Authority{
+						{
+							Key: &KeyRef{KMS: "kms://key/path"},
+							Sources: []Source{
+								{
+									OCI: "registry.example.com",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name:        "Should fail when source oci is empty",
+			expectErr:   true,
+			errorString: "missing field(s): spec.authorities[0].source.oci",
+			policy: ClusterImagePolicy{
+				Spec: ClusterImagePolicySpec{
+					Images: []ImagePattern{{Regex: ".*"}},
+					Authorities: []Authority{
+						{
+							Key: &KeyRef{KMS: "kms://key/path"},
+							Sources: []Source{
+								{
+									OCI: "",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name:        "Should fail when source is empty",
+			expectErr:   true,
+			errorString: "missing field(s): spec.authorities[0].source.oci",
```

Comment:
Umm, but sources could be empty. It could be optionally used.

Reply:
we should only used if present.

Reply:
tl;dr: Test removed.

When I tried to set yaml with
```yaml
spec:
  images:
    - glob: "*"
  authorities:
    - key:
        data: ...
      source:
      - oci: "index.docker.io/dhoang679/nginx-signatures"
      -
```

It responds with:
```
error: error validating "tmp/cip.yaml": error validating data: ValidationError(ClusterImagePolicy.spec.authorities[0].source): unknown object type "nil" in ClusterImagePolicy.spec.authorities[0].source[1]; if you choose to ignore these errors, turn validation off with --validate=false
```

However, this error cant be simulated by unit tests. It did not even enter the validation functions as far as I can tell.

Maybe I misunderstand the `json:"source,omitempty"` as it does not omit when it is an null object from the yaml...

I tried changing the `[]Source` to use a pointer representation `[]*Source` but the unit test with `[]&Source{ nil }` object within did not trigger the same validation error as setting it through yaml.

So went with removing this test for now.